Mathlib/Data/Multiset/Basic.lean

```diff
@@ -528,6 +528,12 @@ theorem le_cons_of_not_mem (m : a ∉ s) : s ≤ a ::ₘ t ↔ s ≤ t := by
     perm_middle.subperm_left.2
       ((subperm_cons _).2 <| ((sublist_or_mem_of_sublist s).resolve_right m₁).subperm)
 
+theorem cons_le_of_not_mem (hs : a ∉ s) (ht : a ∈ t) :
+    a ::ₘ s ≤ t ↔ s ≤ t := by
```

Comment:
```suggestion
    a ::ₘ s ≤ t ↔ a ∈ t ∧ s ≤ t := by
```
and can you prove from the facts that:
* `a ∉ s ↔ Disjoint {a} s`
* `a ::ₘ s = {a} + s`
* `a ∈ t ↔ {a} ≤ t`
* `{a} ∪ s ≤ t ↔ {a} ≤ t ∧ s ≤ t`

Reply:
Sorry, also remove `ht : a ∈ t` from the hypotheses

Reply:
`≤` for Multisets have just been declared around line 500, so there's not much lemmas to use around that line. Check the new commit @YaelDillies

Reply:
> `≤` for Multisets have just been declared around line 500, so there's not much lemmas to use around that line

Why not prove the lemma further down in the file, then?

Reply:
First, there is no need to use more advanced lemmas. It only takes <= 5 lines even without them. Second, L518 declares a similar lemma `le_cons_of_not_mem`, and it is nice for similar lemmas to go together.